test/unit/tool_util/test_test_parsing.py

```diff
@@ -38,3 +38,26 @@ def test_simple_assert_to_test_assert_list():
 
 def test_assert_legacy_same_as_new_list_style():
     assert to_test_assert_list(ASSERT_THAT_LIST) == to_test_assert_list(ASSERT_THAT_LIST)
+
+
+NESTED_ASSERT_LIST = yaml.safe_load(
+    """
+- has_archive_member:
+    path: ".*"
+    children:
```

Comment:
I guess that's what your comment about asserts was about ? It does feel more consistent for that to be

```
asserts:
  - has_text:
    text: thing

? 
```

Reply:
I used the syntax from the other examples in the file

I'm still struggling with yaml, but it seems that 

```python 
yaml.load("""
asserts:
  - has_text:
    text: thing
"""
)

gives `{'asserts': [{'has_text': None, 'text': 'thing'}]}`

while

```
yaml.load("""
asserts:
  - has_text:
      text: thing
"""
)

gives `{'asserts': [{'has_text': {'text': 'thing'}}]}`. Not sure if both versions are parsed at the moment. 

Not sure if the former even works.

Reply:
Sorry, typing without checking, it should be

```
asserts:
  - has_text:
      text: thing
```

Reply:
Yes this is what I meant with my [comment at the planemo issue](https://github.com/galaxyproject/planemo/pull/1417#discussion_r1517559633)

Not sure, but easy to implement I guess. But I can clearly live with the current state. The only question would be if it would simplify anything if we use `asserts`?

Reply:
Yes, then we can re-use models and there's no question what you have to use IMO - it's always asserts.

Reply:
Then remove children completely, or leave it as falll back? I guess since it seems that this was never used, we may just change it?

Reply:
Leave it as a fallback but don't document it would be my suggestion ?